engine/src/main/java/org/terasology/rendering/dag/RenderTaskListGenerator.java

```diff
@@ -227,4 +228,25 @@ public void refresh() {
         generateFrom(nodeList);
     }
 
+    /**
+     * Refreshes the task list if a refresh has been requested.
+     */
+    public void refreshIfNeeded() {
+        if (refreshRequested) {
+            refresh();
+
+            // Clear refresh requests to prevent refreshing unless needed
+            refreshRequested = false;
+        }
+    }
```

Comment:
Thank you @eviltak for the changes. We are nearly there with this. 

I'd prefer that nodes that need a reference to the WorldRenderer get it through injection rather than have it by default, extending the AbstractNode class: there are nodes that do not need a reference to the WorldRenderer  (or won't need it after I refactored them). So I'd revert the changes you did to some nodes and have them look like this:

```
@In
private WorldRenderer worldRenderer
...
worldRenderer.requestTaskListRefresh()
```

Also, I'd like the method ```WorldRenderer.requestRenderTaskListRefresh()``` to be renamed to ```requestTaskListRefresh()```: the renderer has only one task list and the documentation will have explicit references to it as _the_ task list.